src/simulator/cube.py

```diff
@@ -11,29 +10,38 @@ def __init__(self, length: float, height: float, width: float):
         self.height = height
         self.width = width
 
-    def bounding_box(self) -> Box:
+    def bounding_box(self, position: Position) -> Box:
         """Returns smallest enclosing upright Box
-        """
-
-        # TODO: The following calculation is only for an upright cube
-        # dimension calculations
-        length = self.shape.length
-        height = self.shape.height
-        width = self.shape.width
 
-        # position calculations
-        x = self.position.x
-        y = self.position.y
-        z = self.position.z
-        phi = self.position.phi
-        theta = self.position.theta
+           Calculation the smallest bounding box,function moves
+        the shape to the origin position,then rotates it with the
+        corner points. After that,it calculates the dimensions of
+        the box that is tangent to the farthest corners of the rotation.
 
-        # assign return Arguments
-        bb_cube = Cube(length, height, width)
-        bb_position = Position(x, y, z, phi, theta)
-
-        # return Box(1, bb_cube, bb_position)
-        pass
+        """
+        l = self.length
+        w = self.width
+        h = self.height
+        # Notice: Because the shape is symmetrical,
+        # only the corner points of top disk are needed, then double the dimensions.
+        points = [[l / 2, w / 2, h / 2], [-l / 2, w / 2, h / 2],
+                  [l / 2, -w / 2, h / 2], [-l / 2, -w / 2, h / 2]]
+        new_x = list()
+        new_y = list()
+        new_z = list()
+        for point in points:
+            new_point = Shape.rotation(point[0], point[1], point[2], position.phi, position.theta)
+            new_x.append(new_point[0])
+            new_y.append(new_point[1])
+            new_z.append(new_point[2])
+        length = 2 * max([abs(x) for x in new_x])
```

Comment:
why max? why multiplication by 2? try to document what you do in the code. Not as detailed as proposals document but some minimum.